server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -161,18 +196,143 @@ public TextFieldMapper build(BuilderContext context) {
                     builder.fielddataFrequencyFilter(minFrequency, maxFrequency, minSegmentSize);
                     DocumentMapperParser.checkNoRemainingFields(propName, frequencyFilter, parserContext.indexVersionCreated());
                     iterator.remove();
+                } else if (propName.equals("index_prefix")) {
+                    Map<?, ?> indexPrefix = (Map<?, ?>) propNode;
```

Comment:
maybe validate that propNode is indeed not null and a map

Reply:
The block above (for fielddataFrequencyFilter) does no checking here.  I think if somebody passes in `null` explicitly, then returning a NullPointerException is fair enough?

Reply:
I might agree if this was a library but there have been several requests that we produce error messages that are as meaningful as possible so I think it's better to check explicitly?

Reply:
Just added a test for this, and it turns out it's already dealt with by `TypeParsers.parseTextField()` which checks that none of the passed-in parameters are null, so I think we're OK.

Reply:
Perfect. Thanks for checking!